File: 백준/Silver/2164. 카드2/카드2.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int size = scanner.nextInt();
        
        Queue<Integer> queue = new LinkedList<>();
        
        for (int i = 1; i <= size; i++) {
            queue.add(i);
        }
        
        while(queue.size() > 1) {
            int pollNumber = queue.poll();
            int backNumber = queue.poll();
            
            queue.add(backNumber);
        }
        
        System.out.println(queue.poll());
    }
}

File: 백준/Silver/2164. 카드2/카드2.py
# 파이썬에서 queue 문제 풀 때는 deque를 기본으로 쓴다고 생각하기
from collections import deque

size = int(input())
queue = deque(range(1, size + 1))

while len(queue) > 1:
    poll_number = queue.popleft()
    back_number = queue.popleft()
    queue.append(back_number)

print(queue.pop())
